server/src/main/java/org/elasticsearch/index/mapper/TimeSeriesIdFieldMapper.java

```diff
@@ -296,11 +336,18 @@ public void addUnsignedLong(String fieldName, long value) {
             }
         }
 
-        private void add(String fieldName, BytesReference encoded) {
-            BytesReference old = dimensions.put(new BytesRef(fieldName), encoded);
-            if (old != null) {
+        private void add(String fieldName, BytesReference encoded) throws IOException {
+            final BytesArray fieldNameBytesRef = new BytesArray(fieldName);
+            final DimensionDataHolder dimension = new DimensionDataHolder(
+                fieldNameBytesRef,
+                StringHelper.murmurhash3_x86_32(fieldNameBytesRef.toBytesRef(), 0),
```

Comment:
We might consider using hash functions hashing to values with less than 32 bits. Compared to what we usually require from a hashing function (basically no collision unless the values are equal) here what we want is that if two sets of dimensions differ by a very small delta a collision is preferred because it would result in data that is closer to each other.